control/metrics.go

```diff
@@ -55,17 +55,28 @@ var (
 
 // hostnameReader, hostnamer created for mocking
 func init() {
-	hostnameReader = &hostnameReaderType{}
+	h, err := os.Hostname()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"_module": "control",
+			"_file":   "metrics.go,",
+			"_block":  "addStandardAndWorkflowTags",
+			"error":   err.Error(),
+		}).Error("Unable to determine hostname")
+	}
+	hostnameReader = &hostnameReaderType{hostname: h}
```

Comment:
@lynxbat I believe so, I can move the hostname out into the metrics.go file and have it run every collection. This would also have to be done at [control/plugin_manager.go](https://github.com/intelsdi-x/snap/blob/master/control/plugin_manager.go)